### PR TITLE
Support for collection in Tracker

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Fixes https://github.com/PyTorchLightning/metrics/issues/681
Enable the option to pass into `MetricCollection` the wrapper of `MetricTracker`.

## Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by justusschock — APPROVED** [review]

**Comment by Borda** [comment]
@SkafteNicki is this ready to go? :]

File: tests/wrappers/test_tracker.py [hunk 1]
```diff
@@ -17,16 +17,25 @@
...
-from torchmetrics import Accuracy, MeanAbsoluteError, MeanSquaredError, Precision, Recall
+from torchmetrics import Accuracy, MeanAbsoluteError, MeanSquaredError, MetricCollection, Precision, Recall
 from torchmetrics.wrappers import MetricTracker
 
 seed_all(42)
 
 
 def test_raises_error_on_wrong_input():
-    with pytest.raises(TypeError, match="metric arg need to be an instance of a torchmetrics metric .*"):
+    """make sure that input type errors are raised on wrong input."""
```
Comment (Borda) [inline tests/wrappers/test_tracker.py:27]:
```suggestion
    """Make sure that input type errors are raised on the wrong input."""
```

File: torchmetrics/wrappers/tracker.py [hunk 2]
```diff
@@ -96,7 +132,13 @@ def compute(self) -> Any:
...
         """Compute the metric value for all tracked metrics."""
         self._check_for_increment("compute_all")
-        return torch.stack([metric.compute() for i, metric in enumerate(self) if i != 0], dim=0)
+        # The i!=0 accounts for the self._base_metric should be ignored
+        res = [metric.compute() for i, metric in enumerate(self) if i != 0]
+        if isinstance(self._base_metric, MetricCollection):
+            keys = res[0].keys()
+            return {k: torch.stack([r[k] for r in res], dim=0) for k in keys}
+        else:
+            return torch.stack(res, dim=0)
```
Comment (Borda) [inline torchmetrics/wrappers/tracker.py:141]:
```suggestion
        return torch.stack(res, dim=0)
```

File: torchmetrics/wrappers/tracker.py [hunk 3]
```diff
@@ -55,15 +56,50 @@ class MetricTracker(nn.ModuleList):
...
+        current stats={'MeanSquaredError': tensor(1.9800), 'ExplainedVariance': tensor(-0.9199)}
+        current stats={'MeanSquaredError': tensor(2.2481), 'ExplainedVariance': tensor(-1.1622)}
+        >>> best_res, which_epoch = tracker.best_metric(return_step=True)
+        >>> best_res
+        {'MeanSquaredError': 1.8218144178390503, 'ExplainedVariance': -0.8297995328903198}
+        >>> which_epoch
+        {'MeanSquaredError': 0, 'ExplainedVariance': 2}
+        >>> tracker.compute_all()  # doctest: +NORMALIZE_WHITESPACE
+        {'MeanSquaredError': tensor([1.8218, 2.0268, 1.9491, 1.9800, 2.2481]),
+         'ExplainedVariance': tensor([-0.8969, -1.0206, -0.8298, -0.9199, -1.1622])}
```
Comment (Borda) [inline torchmetrics/wrappers/tracker.py:88]:
```suggestion
        >>> from pprint import pprint
        >>> best_res, which_epoch = tracker.best_metric(return_step=True)
        >>> pprint(best_res)
        {'MeanSquaredError': 1.8218144178390503, 'ExplainedVariance': -0.8297995328903198}
        >>> which_epoch
        {'MeanSquaredError': 0, 'ExplainedVariance': 2}
        >>> pprint(tracker.compute_all())
        {'MeanSquaredError': tensor([1.8218, 2.0268, 1.9491, 1.9800, 2.2481]),
         'ExplainedVariance': tensor([-0.8969, -1.0206, -0.8298, -0.9199, -1.1622])}
```

File: torchmetrics/wrappers/tracker.py [hunk 4]
```diff
@@ -55,15 +56,50 @@ class MetricTracker(nn.ModuleList):
         current acc=0.07999999821186066
         current acc=0.10199999809265137
         >>> best_acc, which_epoch = tracker.best_metric(return_step=True)
+        >>> best_acc
+        0.12600000202655792
```
Comment (Borda) [inline torchmetrics/wrappers/tracker.py:60]:
```suggestion
        >>> best_acc  # doctest: +ELLIPSIS
        0.1260...
```

**Review by Borda — APPROVED** [review]
lgtm :rabbit:

**Pull request merged**